lib/views/pages/home_page.dart

```diff
@@ -0,0 +1,60 @@
+import "package:flutter/material.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/utils/ui/circular_value.dart";
+import "package:proxima/viewmodels/login_view_model.dart";
+import "package:proxima/viewmodels/profile_view_model.dart";
+import "package:proxima/views/navigation/routes.dart";
+
+class HomePage extends HookConsumerWidget {
+  static const logoutButtonKey = Key("logout");
+
+  const HomePage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Home page :)"),
+      ),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            const UserProfile(),
+            const SizedBox(
+              height: 20,
+            ),
+            ElevatedButton(
+              key: logoutButtonKey,
+              onPressed: () {
+                ref.read(loginServiceProvider).signOut();
+
+                // Go to login page when the user is logged out
+                Navigator.pushNamedAndRemoveUntil(
```

Comment:
1. Why using this instead of `Navigator.pushReplacementNamed(context, Routes.login.name);`?
2. Is it possible that the user gets logged out without clicking this button? For instance, if they loose their connection, or change their password while being logged in? In case this possible, would there be a possibility to listen to `FirebaseAuth.instance.authStateChanges()` and change the route in case the user is not logged in?

Reply:
1. In case, the user is on another page while getting logged out.
2. Yes, see https://github.com/ProximaEPFL/proxima/pull/25#discussion_r1532986804